grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -186,6 +186,9 @@ class InvoiceService {
         InvoiceItem invoiceItem = InvoiceItem.get(id)
         Invoice invoice = invoiceItem.invoice
         invoice.removeFromInvoiceItems(invoiceItem)
+        invoiceItem.orderItems?.each { OrderItem it -> it.removeFromInvoiceItems(invoiceItem) }
+        invoiceItem.orderAdjustments?.each { OrderAdjustment it -> it.removeFromInvoiceItems(invoiceItem) }
+        invoiceItem.shipmentItems?.each { ShipmentItem it -> it.removeFromInvoiceItems(invoiceItem) }
```

Comment:
Move this to a new method. Add some error handling if it makes you feel better i.e. null check on invoiceItem
```
void deleteInvoiceItem(InvoiceItem invoiceItem) {
    invoiceItem.orderItems?.each { OrderItem it -> it.removeFromInvoiceItems(invoiceItem) }
    invoiceItem.orderAdjustments?.each { OrderAdjustment it -> it.removeFromInvoiceItems(invoiceItem) }
    invoiceItem.shipmentItems?.each { ShipmentItem it -> it.removeFromInvoiceItems(invoiceItem) }
    invoiceItem.delete()
}
```

Reply:
done